BrewMesModulesParent/schedule/src/test/java/com/brewmes/schedule/ScheduleServiceImplTest.java

```diff
@@ -0,0 +1,184 @@
+package com.brewmes.schedule;
+
+import com.brewmes.common.entities.ScheduledBatch;
+import com.brewmes.common.util.Products;
+import com.brewmes.common_repository.ScheduleRepository;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+class ScheduleServiceImplTest {
+
+    private static ScheduledBatch scheduledBatch;
+    private static ScheduledBatch scheduledBatch2;
+    private static List<ScheduledBatch> scheduledBatches;
+    private static List<ScheduledBatch> scheduledBatchesRemoved;
+    private static List<ScheduledBatch> emptyScheduledBatches;
+    private static List<ScheduledBatch> prioList;
+    @Mock
+    ScheduleRepository scheduleRepository;
+    @InjectMocks
+    ScheduleServiceImpl scheduleService;
+
+    @BeforeAll
+    static void setUp() {
+        scheduledBatch = new ScheduledBatch(100, Products.PILSNER, 200);
+        scheduledBatch2 = new ScheduledBatch(80, Products.PILSNER, 100);
+
+        scheduledBatches = new ArrayList<>();
+        scheduledBatches.add(scheduledBatch);
+        scheduledBatches.add(scheduledBatch2);
+
+        scheduledBatchesRemoved = new ArrayList<>();
+        scheduledBatchesRemoved.add(scheduledBatch2);
+
+        emptyScheduledBatches = new ArrayList<>();
+
+        prioList = new ArrayList<>();
+        prioList.add(scheduledBatch2);
+        prioList.add(scheduledBatch);
+    }
+
+    @Test
+    void addToQueue_success() {
+        when(scheduleRepository.insert(scheduledBatch)).thenReturn(scheduledBatch);
+        when(scheduleRepository.findAll()).thenReturn(scheduledBatches);
+
+        int expected = scheduleRepository.findAll().indexOf(scheduledBatch);
+        int actual = scheduleService.addToQueue(scheduledBatch);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void addToQueue_failure() {
+        when(scheduleRepository.findAll()).thenReturn(scheduledBatches);
+
+        int expected = scheduleRepository.findAll().indexOf(scheduledBatch2);
```

Comment:
Skal vel hellere bare hard codes så vi ikke bare i bund og grund kalder metoden to gange?